File: backend/src/ShaktiUdyog.Domain/Entities/Company.cs
namespace ShaktiUdyog.Domain.Entities;

/// <summary>
/// Customer company (requirements §14). Customer users access records only
/// through an approved UserCompany link to a company.
/// </summary>
public class Company
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? LegalBusinessName { get; set; }
    public string? BusinessType { get; set; }
    public string? Industry { get; set; }
    public string? Website { get; set; }
    public string? CompanyEmail { get; set; }
    public string? CompanyPhone { get; set; }
    public string? PurchaseEmail { get; set; }
    public string? AccountsEmail { get; set; }
    public string? AddressLine1 { get; set; }
    public string? FactoryAddress { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? PostalCode { get; set; }
    public string? PinCode { get; set; }
    public string? Country { get; set; }
    public string? GstNumber { get; set; }
    public string? PANNumber { get; set; }
    public string? CINNumber { get; set; }
    public string? MSMENumber { get; set; }
    public string? PreferredCurrency { get; set; }
    public string? PreferredPaymentMethod { get; set; }
    public string? PreferredCommunication { get; set; }
    public string? PreferredLanguage { get; set; }
    public string? CompanyLogoUrl { get; set; }
    public string? VerificationStatus { get; set; } = "Pending";
    public string? DeliveryAddresses { get; set; }
    public DateTimeOffset? VerificationSubmittedOn { get; set; }
    public DateTimeOffset? VerifiedOn { get; set; }
    public Guid? VerifiedByUserId { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    // Navigation
    public ApplicationUser? VerifiedBy { get; set; }
    public List<ContactPerson> ContactPersons { get; set; } = [];
    public List<CompanyAddress> Addresses { get; set; } = [];
    public List<CompanyDocument> Documents { get; set; } = [];
}

File: frontend/src/types/company.ts
/**
 * Company profile, addresses, and corporate account domain models.
 */

export interface CompanyAddress {
  id: string;
  label: string;
  addressLine1: string;
  addressLine2?: string | null;
  city: string;
  state: string;
  postalCode: string;
  country: string;
  isDefault: boolean;
}

export interface CompanyDetail {
  id: string;
  name: string;
  legalBusinessName: string | null;
  businessType: string | null;
  industry: string | null;
  website: string | null;
  companyEmail: string | null;
  companyPhone: string | null;
  purchaseEmail: string | null;
  accountsEmail: string | null;
  registeredAddress: string | null;
  factoryAddress: string | null;
  city: string | null;
  state: string | null;
  postalCode: string | null;
  country: string | null;
  gstNumber: string | null;
  panNumber: string | null;
  tanNumber: string | null;
  cinNumber: string | null;
  msmeNumber: string | null;
  bankName: string | null;
  bankAccountNumber: string | null;
  bankIfscCode: string | null;
  bankBranch: string | null;
  accountType: string | null;
  isApproved: boolean;
  creditLimit: number | null;
  paymentTermsDays: number | null;
  addresses?: CompanyAddress[];
  contacts?: ContactPerson[];
  documents?: CompanyDocument[];
}

export interface ContactPerson {
  id: string;
  name: string;
  designation: string | null;
  email: string | null;
  phone: string | null;
  isPrimary: boolean;
}

export interface CompanyDocument {
  id: string;
  documentType: string;
  fileName: string;
  fileSizeBytes: number;
  uploadedAtUtc: string;
}

export interface Profile {
  email: string;
  fullName: string | null;
  phoneNumber: string | null;
  company: {
    name: string;
    addressLine1: string | null;
    city: string | null;
    state: string | null;
    postalCode: string | null;
    country: string | null;
    gstNumber: string | null;
    deliveryAddresses: string | null;
  } | null;
  mfaEnabled: boolean;
  accountCreatedAtUtc: string | null;
}
